### PR TITLE
MWPW-136264 : Blog URL redirects automation test script

### PR DESCRIPTION
**Pull request by skumar09**
[MWPW-136264](https://jira.corp.adobe.com/browse/MWPW-136264): Blog URL redirects automation test script

PR includes 
- Automation test scripts for the following Blog redirect URLs

  -  Topic URL Redirects,
  -  Index URL Redirects,
  -  Author URL Redirects,
  -  Publish URL Redirects,

### CODE REVIEW

**Review by amauch-adobe — CHANGES_REQUESTED** [review]
Would it be best to make a generic URL redirect test that is one method but takes in a data file instead so the test can be used against many different sites instead of just blog.adobe.com?

**Comment by skumar09** [comment]
> Would it be best to make a generic URL redirect test that is one method but takes in a data file instead so the test can be used against many different sites instead of just blog.adobe.com?

@amauch-adobe: That makes sense, and redirect validation is a generic scenario across apps. Let me work on that. 
I need this PR to go through for any quick blog regression.

File: tests/blog/urlredirect.test.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      stepCounter = 0;
+      errors.length = 0;
+
+      for (const [url, redirectUrl] of Object.entries(path)) {
+        try {
+          await test.step(`step-${++stepCounter}: verify url '${url}' redirect`, async () => {
+            console.info(`[Test Page ${stepCounter}]: ${baseURL}${url}`);
+            console.info(`[Redirect Url]: ${redirectBlogUrl}${redirectUrl}\n`);
+            await page.goto(`${baseURL}${url}`, { waitUntil: 'domcontentloaded' });
+            await expect(page).toHaveURL(`${redirectBlogUrl}${redirectUrl}`);
```
Comment (JackySun9) [inline tests/blog/urlredirect.test.js:21]: you can use soft expect

Comment (skumar09) [reply tests/blog/urlredirect.test.js]: @JackySun9: Yes, In this case, soft assertions are simple and a better choice than having a custom code to catch errors

File: tests/blog/urlredirect.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      const path = features[1].path;
+      stepCounter = 0;
+      errors.length = 0;
+
+      for (const [url, redirectUrl] of Object.entries(path)) {
+        await test.step(`step-${++stepCounter}: verify url '${url}' redirect`, async () => {
+          console.info(`[Test Page ${stepCounter}]: ${baseURL}${url}`);
+          console.info(`[Redirect Url]: ${redirectBlogUrl}${redirectUrl}\n`);
+          await page.goto(`${baseURL}${url}`, { waitUntil: 'domcontentloaded' });
+          await expect(page).toHaveURL(`${redirectBlogUrl}${redirectUrl}`);
```
Comment (JackySun9) [inline tests/blog/urlredirect.test.js:36]: soft expect

File: tests/blog/urlredirect.test.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      const path = features[2].path;
+      stepCounter = 0;
+      errors.length = 0;
+
+      for (const [url, redirectUrl] of Object.entries(path)) {
+        await test.step(`step-${++stepCounter}: verify url '${url}' redirect`, async () => {
+          console.info(`[Test Page ${stepCounter}]: ${baseURL}${url}`);
+          console.info(`[Redirect Url]: ${redirectBlogUrl}${redirectUrl}\n`);
+          await page.goto(`${baseURL}${url}`, { waitUntil: 'domcontentloaded' });
+          await expect(page).toHaveURL(`${redirectBlogUrl}${redirectUrl}`);
```
Comment (JackySun9) [inline tests/blog/urlredirect.test.js:52]: same here

File: tests/blog/urlredirect.test.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      const path = features[3].path;
+      stepCounter = 0;
+      errors.length = 0;
+
+      for (const [url, redirectUrl] of Object.entries(path)) {
+        await test.step(`step-${++stepCounter}: verify url '${url}' redirect`, async () => {
+          console.info(`[Test Page ${stepCounter}]: ${baseURL}${url}`);
+          console.info(`[Redirect Url]: ${redirectBlogUrl}${redirectUrl}\n`);
+          await page.goto(`${baseURL}${url}`, { waitUntil: 'domcontentloaded' });
+          await expect(page).toHaveURL(`${redirectBlogUrl}${redirectUrl}`);
```
Comment (JackySun9) [inline tests/blog/urlredirect.test.js:68]: soft expect

**Review by JackySun9 — APPROVED** [review]

**Pull request merged**